nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FetchFileTransfer.java

```diff
@@ -296,6 +298,18 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             } else {
                 attributes.put(CoreAttributes.FILENAME.key(), filename);
             }
+
+            if (failureRelationship != null) {
+                attributes.put(FAILURE_REASON_ATTRIBUTE, failureRelationship.getName());
+                flowFile = session.putAllAttributes(flowFile, attributes);
+                session.transfer(session.penalize(flowFile), failureRelationship);
+                if (provenanceEventOnFailure) {
+                    session.getProvenanceReporter().route(flowFile, failureRelationship);
+                }
+                cleanupTransfer(transfer, closeConnOnFailure, transferQueue, host, port);
+                return;
+            }
```

Comment:
Moving this logic to a separate block outside of the catch blocks introduces some potential logic changes, such as the addition of other attributes to the FlowFile, is that intentional? Otherwise it would be less of an impact to apply the attribute value in the appropriate catch blocks.

Reply:
Yes, this was intentional.  I would think the other attributes are needed for a follow on processor to have context (ex: generate email notifications with similar content to what is being logged in the catch blocks).  Also, this was meant to remove some redundant code.

Reply:
Thanks, that makes sense, and sounds good.